frontend/src/routes/Home/Overview/OverviewPageBeta.tsx

```diff
@@ -99,10 +101,58 @@ export default function OverviewPageBeta(props: { selectedClusterLabels: Record<
   const [placementDecisions] = useRecoilState(placementDecisionsState)
   const [policyReports] = useRecoilState(policyreportState)
   const [subscriptions] = useRecoilState(subscriptionsState)
-  const [isInsightsSectionOpen, setIsInsightsSectionOpen] = useState<boolean>(true)
   const [isClusterSectionOpen, setIsClusterSectionOpen] = useState<boolean>(true)
+  const [isInsightsSectionOpen, setIsInsightsSectionOpen] = useState<boolean>(true)
+  const [isObservabilityQueryLoading, setIsObservabilityQueryLoading] = useState<boolean>(true)
+  const [isObservabilityInstalled, setIsObservabilityInstalled] = useState<boolean>(false)
+  const [grafanaRoute, setGrafanaRoute] = useState('')
   GetDiscoveredOCPApps(applicationsMatch.isExact, !ocpApps.length && !discoveredApplications.length)
 
+  // Query to determine if Observability is installed
+  useEffect(() => {
+    getResource({
+      apiVersion: 'observability.open-cluster-management.io/v1beta2',
+      kind: 'MultiClusterObservability',
+      metadata: {
+        name: 'observability',
+      },
+    })
+      .promise.then((response: any) => {
+        if (response?.status?.conditions as V1CustomResourceDefinitionCondition[]) {
+          const isRunning = response.status.conditions.find(
+            (condition: { lastTransitionTime: Date; message: string; reason: string; status: string; type: string }) =>
+              condition.message === 'Observability components are deployed and running'
+          )
+          setIsObservabilityInstalled(isRunning !== undefined)
+          setIsObservabilityQueryLoading(false)
+        }
+      })
+      .catch((err) => {
+        console.error('Error getting MultiClusterObservability resource: ', err)
+        setIsObservabilityQueryLoading(false)
+      })
+  }, [])
+
+  // Query for grafana Route
+  useEffect(() => {
+    getResource({
+      apiVersion: 'route.openshift.io/v1',
+      kind: 'Route',
+      metadata: {
+        name: 'grafana',
+        namespace: 'open-cluster-management-observability',
+      },
+    })
+      .promise.then((route: any) => {
+        if (route?.spec?.host) {
+          setGrafanaRoute(`${route?.spec?.tls?.termination ? 'https' : 'http'}://${route.spec.host}`)
+        }
+      })
+      .catch((err) => {
+        console.error('Error getting Observability Route: ', err)
+      })
+  }, [])
+
```

Comment:
RBAC concerns here - not all users will be able to see these resources.

Reply:
Good point - Looks like Clusters page gets the grafana link via the `observability-controller` ClusterManagementAddOn resource. I can switch this to do the same. Should also make the page load faster by using recoil state.

Reply:
Even via recoil, the resources are filtered based on the user's permissions, but I guess we have this issue anywhere we show the Grafana link currently.